articles/regular-expressions-in-python/index.md

```diff
@@ -0,0 +1,269 @@
+
+    ```py
+    test_string = "I d0n't want any num6er 1n th1s 5tr1ng"
+    regex = re.findall("[0-9]",test_string)
+    if regex:
+        print("There is(are) number(s) in the string")
+    else:
+        print("There are no numbers in this string")
+
+    # Output - There is(are) number(s) in the string
+    ```
+
+2. Another rule that metacharacters enforce is that they do not have any effect in a character class unless it starts with a metacharacter. A complement of a character class is denoted by the metacharacter "^". So, "[^abc]" will match with any character except "a", "b" and "c" whereas "[abc^]" will match with characters "a", "b", "c" and "^" even though "^" is a metacharacter. Note that the complement happens only within a character class.
+
+    ```py
+    test_string = "This string has no e in it"
+    regex = re.search("[^e]",test_string)
+    if regex:
+        print("There is an e in the string")
+    else:
+        print("There is no e in this string")
+
+    # Output - There is an e in the string
+
+    test_string = "2^4 equals 16"
+    regex = re.search("[z^]",test_string)
+    if regex:
+        print("There is either a z or a ^ in this string")
+    else:
+        print("There is no z or ^ in this string")
+
+    # Output - There is either a z or a ^ in this string
+    ```
+
+3. A combination of character classes is allowed in RegExes. "[0-2][0-4]" matches with any number between 00 and 24, "[ac][b-d]" matches with the strings "ab", "ac", "ad", "cb", "cc" and "cd", and "[a-zA-Z]" returns a match for any character between a to z, lowercase or uppercase. The same extends to numbers as well.
+
+    ```py
+    test_string = "Brutus stabbed Caesar"
+    regex = re.search("[aeiou][aeiou]",test_string)
+    if regex:
+        print("There are consecutive vowels in this string")
+    else:
+        print("There are no consecutive vowels in this string")
+
+    # Output - There are consecutive vowels in this string
+
+    print(regex)
+
+    # <_sre.SRE_Match object; span=(16, 18), match='ae'>
+    ```
+
+    The span parameter in the RegEx object displays the position of the match (in this case, "ae").
+
+4. "^" within a string matches with a string that starts with the sequence of characters after "^" and "$" within a string matches with a string that ends with the sequence of characters before "$".
+
+    ```py
+    names = ["Steve Aoki","Joe Willink","Steve Carrel","Jim Ross"]
+    for name in names:
+        if re.search("^Steve",name):
+            print(name)
+
+    # Output - Steve Aoki
+    #          Steve Carrel
+
+    names = ["Evan Smith","Gary Neville","Jamie Carragher","Steve Smith","Elaine Smith"]
+    names_starting_with_steve = [name for name in names if re.search("Smith$",name)]
```

Comment:
`names_ending_with_smith` would be a more appropriate variable name

Reply:
Hey @adrianmurage, thanks for the review! I've made the changes you've asked for.